src/agents/workflows/subgraphs/step2_router/graph.py

```
@@ -299,15 +360,30 @@ def intent_route(state: AgentState) -> Literal["Planner", "Executor"]:
         return "Planner"
 
     _, _, combined_question = _collect_user_context(state)
+
+    current_chunks = state.get("solution_chunks") or []
+    target_num = state.get("target_problem_number")
+    last_idx = state.get("last_solved_index")
+
+    # [개선] 2. 특정 문제를 지칭했거나, 이미 풀이 루프 안에 있다면 Planner를 건너뛰고 Executor로 직행
+    if target_num is not None or last_idx is not None:
+        state["plan"] = ["Solve"]
+        print(f"---ROUTER: CONTINUING LOOP OR TARGETING -> GO TO EXECUTOR---")
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)